lib/ui/wallet/usecases/get_transaction_history_data_use_case.dart

```diff
@@ -19,14 +20,41 @@ class GetTransactionHistoryDataUseCase {
     this._authRepository,
   );
 
-  Future<Result<List<WalletTransactionTileData>, HyphaError>> run(bool transferOnly) async {
+  Future<Result<List<WalletTransactionTileData>, HyphaError>> getTransferTransactionsForTokens() async {
     final List<FirebaseTokenData> allTokens = await _firebaseDatabaseService.getAllTokens();
     final user = _authRepository.authDataOrCrash;
 
-    final result = await _getTransactionHistoryUseCase.run(transferOnly);
+    final result = await _getTransactionHistoryUseCase.run(true);
     if (result.isValue) {
       return Result.value(result.valueOrCrash
-          .where((TransactionModel element) => element is TransactionRedeem || element is TransactionTransfer)
+          .where(
+            (TransactionModel element) =>
+                element is TransactionRedeem ||
+                element is TransactionTransfer &&
+                    allTokens.firstWhereOrNull(
+                            (token) => token.contract == element.account && token.symbol == element.symbol) !=
+                        null,
+          )
+          .toList()
+          .mapRecentTransactionsTileData(allTokens, user.userProfileData.accountName));
+    } else {
+      return Result.error(result.asError!.error);
+    }
+  }
+
+  Future<Result<List<WalletTransactionTileData>, HyphaError>> getTransferTransactionsForToken({
```

Comment:
We can now chose to show transactions for all tokens we support. or only for 1 token the user picks. Token Details page